File: backend/src/schemas/recommendations.py
"""
Recommendation response schemas
"""

from __future__ import annotations

from datetime import date, datetime
from typing import Any
from uuid import UUID

from pydantic import AnyUrl, BaseModel, Field


class FlightOptionSchema(BaseModel):
    """Flight option detail"""

    id: UUID
    provider: str
    carrier: str
    flight_number: str
    departure_airport: str
    arrival_airport: str
    departure_time: datetime
    arrival_time: datetime
    duration_minutes: int
    stops: int
    seat_class: str | None = None
    baggage_info: dict[str, Any] | None = None
    price_currency: str
    price_amount: int
    booking_url: AnyUrl
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class FlightRecommendationsResponse(BaseModel):
    """Flight recommendation payload"""

    plan_id: UUID
    origin_airport: str
    destination_airport: str
    options: list[FlightOptionSchema] = Field(default_factory=list)


class AccommodationOptionSchema(BaseModel):
    """Accommodation option detail"""

    id: UUID
    provider: str
    name: str
    description: str | None = None
    address: str | None = None
    city: str | None = None
    country: str | None = None
    rating: float | None = None
    review_count: int | None = None
    star_rating: float | None = None
    price_currency: str
    price_per_night: int | None = None
    total_price: int
    check_in_date: date | None = None
    check_out_date: date | None = None
    nights: int | None = None
    room_type: str | None = None
    booking_url: AnyUrl
    image_url: AnyUrl | None = None
    amenities: list[str] | None = None
    tags: list[str] | None = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class AccommodationRecommendationsResponse(BaseModel):
    """Accommodation recommendation payload"""

    plan_id: UUID
    check_in: date | None = None
    check_out: date | None = None
    options: list[AccommodationOptionSchema] = Field(default_factory=list)

File: shared/types/recommendations.ts
export interface FlightOption {
  id: string;
  provider: string;
  carrier: string;
  flight_number: string;
  departure_airport: string;
  arrival_airport: string;
  departure_time: string;
  arrival_time: string;
  duration_minutes: number;
  stops: number;
  seat_class?: string | null;
  baggage_info?: Record<string, unknown> | null;
  price_currency: string;
  price_amount: number;
  booking_url: string;
  created_at: string;
  updated_at: string;
}

export interface FlightRecommendations {
  plan_id: string;
  origin_airport: string;
  destination_airport: string;
  options: FlightOption[];
}

export interface AccommodationOption {
  id: string;
  provider: string;
  name: string;
  description?: string | null;
  address?: string | null;
  city?: string | null;
  country?: string | null;
  rating?: number | null;
  review_count?: number | null;
  star_rating?: number | null;
  price_currency: string;
  price_per_night?: number | null;
  total_price: number;
  check_in_date?: string | null;
  check_out_date?: string | null;
  nights?: number | null;
  room_type?: string | null;
  booking_url: string;
  image_url?: string | null;
  amenities?: string[] | null;
  tags?: string[] | null;
  created_at: string;
  updated_at: string;
}

export interface AccommodationRecommendations {
  plan_id: string;
  check_in?: string | null;
  check_out?: string | null;
  options: AccommodationOption[];
}
